### DISCUSSION TITLE
Dedicated Workplane Panel

### DISCUSSION THREAD

**Discussion by Intuos5** [hlorus/CAD_Sketcher#99] [Feature Requests]
**Motivation / Existing Problem**
Currently, all solvespace points, including workplanes are shown in the same panel under "Entities". This makes them hard to differentiate from the constraint solver points. 

**Description**
 If the workplanes were moved to a dedicated panel category, called "workplanes" it would be easier to show/ hide specific workplanes. It would also allow for changing workplanes to a different face more easily. Ideally, workplanes should also be renameable.

**Comment by TheMakerTales** [comment]
I quite like this idea too, if not this  then maybe having workplanes show up as 1 entiti instead of 4/7

**Comment by hlorus** [comment]
I would suggest to have a filter for the entity browser and/or list entities with an hierarchy where points wouldn't be displayed on their own but as a child of the lines/arcs/cicles that depend on them.

**Reply by Arch-druid** [reply]
This is exactly what I mentioned in my feature request here!

https://github.com/hlorus/CAD_Sketcher/discussions/181

This absolutely needs to happen so that the entities are not simply splattered all over a  dialogue box!

**Comment by Intuos5** [comment]
I agree with that @hlorus 
But that still does not negate the need for a dedicated workplane viewer, because it would still take several clicks to filter with buttons IMO.

**Reply by hlorus** [reply]
Filter options could be stored in the addon preferences, that way you'd have to set it only once. The default filter settings could even be adjusted to make more sense for the general use.

IMO workplanes are basic entities like the others, i don't really see why they would need any special treatment. Usually one creates a workplane to then immediately place a sketch on it, after that the main interaction is with the sketch(which is already displayed in a specific browser).

**Reply by Intuos5** [reply]
Maybe that's because I envision workplanes to be used like Fusion 360, where you can adjust the workplanes and affect the way the geometry is created. That functionality may or may not be part of the addon, in which case there doesn't need to be such a panel. FYI in Fusion, you can also reproject sketches to another workplane and they can always be toggled on from the model browser, which displays the workplanes per 'component' (i.e. object).
